arangod/IResearch/IResearchViewCoordinator.cpp

```diff
@@ -340,11 +340,54 @@ bool IResearchViewCoordinator::emplace(
     uint64_t planVersion,
     LogicalView::PreCommitCallback const& preCommit
 ) {
+  auto& properties = info.isObject() ? info : emptyObjectSlice(); // if no 'info' then assume defaults
+  std::string error;
+
+  bool hasLinks = properties.hasKey("links");
+
   auto view = std::shared_ptr<IResearchViewCoordinator>(
     new IResearchViewCoordinator(vocbase, info, planVersion)
   );
-  auto& properties = info.isObject() ? info : emptyObjectSlice(); // if no 'info' then assume defaults
-  std::string error;
+
+  if (hasLinks) {
+    auto* engine = arangodb::ClusterInfo::instance();
+
+    if (!engine) {
+      return nullptr;
+    }
+
+    // check link auth as per https://github.com/arangodb/backlog/issues/459
+    if (arangodb::ExecContext::CURRENT) {
+      // check existing links
+      for (auto& entry: view->_collections) {
```

Comment:
why do you expect there to be collections present?

Reply:
This is just copy past error.